File: ankiety-backend/Surveys.Core/Entities/Invitation.cs
using System;

namespace Surveys.Core.Entities
{
    public class Invitation
    {
        public Guid Id { get; set; }

        public string UserId { get; set; }

        public Guid SurveyId { get; set; }

        public DateTime SendDate { get; set; }

        public DateTime? StartDate { get; set; }

        public DateTime? ExpirationDate { get; set; }

        public DateTime? FilledDate { get; set; }

        public virtual User User { get; set; }

        public virtual Survey Survey { get; set; }
    }
}

File: ankiety-frontend/src/app/models/invitation.ts
export class Invitation {
  constructor(
    public id: string,
    public userId: string,
    public surveyId: string,
    public sendDate: string,
    public startDate: string,
    public expirationDate: string,
    public questionsOnPage: number,
    public username: string,
    public surveyName: string,
    public expired: boolean,
    public filledSurvey: boolean,
  ) { }
}
